sdk/identity/azure_identity/src/credentials/imds_managed_identity_credentials.rs

```diff
@@ -75,6 +76,7 @@ impl ImdsManagedIdentityCredential {
             secret_env: secret_env.to_owned(),
             id,
             cache: TokenCache::new(),
+            env: options.env().clone(),
```

Comment:
This is already in `TokenCredentialOptions`, though. Can you not just use that?

Reply:
I mean, don't declare it when you already have it in the nested options. Just use what's already defined.

Reply:
Please forgive my density here, I'm still not following you. I believe the struct needs a new field because it doesn't currently hold the options. I chose to add the Env to the struct instead of the options because that's a smaller change. Really the right thing to do would be to read environment variables in the constructor and hang on to those values for use in `get_token`, however I don't want to jump into a rewrite yet. With this PR, I just want to put a better API on the current implementation

Reply:
Ah, sorry. My bad. The way GitHub was rendering the diff and not the whole file, I thought the struct was holding the `options` already.